File: ApiCall/OllamaAPI.java
import java.io.*;
import java.net.*;
import java.nio.charset.StandardCharsets;
import java.text.SimpleDateFormat;
import java.util.*;

import org.json.JSONObject;

public class OllamaAPI {
    public static void main(String[] args) {
        // Default parameters
        String hostApi = "http://localhost:11434";
        String model = "phi3:mini";
        String prompt = "What is the capital of France?";
        boolean printRaw = false;
        boolean enableLogging = false;

        // Parse command-line arguments
        for (String arg : args) {
            if (arg.startsWith("--hostApi=")) hostApi = arg.substring(10);
            else if (arg.startsWith("--model=")) model = arg.substring(8);
            else if (arg.startsWith("--prompt=")) prompt = arg.substring(9);
            else if (arg.equals("--printRaw")) printRaw = true;
            else if (arg.equals("--enableLogging")) enableLogging = true;
        }

        String logFile = "llm.log";

        // Logging function
        Runnable logInit = () -> {
            File f = new File(logFile);
            if (!f.exists()) try { f.createNewFile(); } catch (IOException ignored) {}
        };
        logInit.run();

        voidLog("Run: hostApi='" + hostApi + "', model='" + model + "', prompt='" + prompt + "', printRaw=" + printRaw, enableLogging, logFile);

        try {
            // Prepare JSON body
            JSONObject bodyJson = new JSONObject();
            bodyJson.put("model", model);
            bodyJson.put("prompt", prompt);
            String body = bodyJson.toString();

            // HTTP POST
            URL url = new URL(hostApi + "/api/generate");
            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
            conn.setRequestMethod("POST");
            conn.setRequestProperty("Content-Type", "application/json");
            conn.setDoOutput(true);
            try (OutputStream os = conn.getOutputStream()) {
                byte[] input = body.getBytes(StandardCharsets.UTF8);
                os.write(input, 0, input.length);
            }

            int statusCode = conn.getResponseCode();
            voidLog("STATUS CODE: " + statusCode, enableLogging, logFile);
            System.out.println("STATUS CODE: " + statusCode);

            InputStream is = (statusCode >= 200 && statusCode < 400) ? conn.getInputStream() : conn.getErrorStream();
            String responseString = new String(is.readAllBytes(), StandardCharsets.UTF8);

            StringBuilder fullResponse = new StringBuilder();
            for (String line : responseString.split("\n")) {
                if (!line.trim().isEmpty()) {
                    try {
                        JSONObject json = new JSONObject(line);
                        if (json.has("response")) {
                            fullResponse.append(json.getString("response"));
                        }
                    } catch (Exception ignored) {}
                }
            }

            if (printRaw) {
                System.out.println(responseString);
                voidLog("RAW RESPONSE: " + safeSub(responseString, 200) + "...", enableLogging, logFile);
            } else {
                System.out.println(fullResponse);
                voidLog("FULL RESPONSE: " + safeSub(fullResponse.toString(), 200) + "...", enableLogging, logFile);
            }
        } catch (Exception e) {
            System.out.println("ERROR:");
            e.printStackTrace();
            voidLog("ERROR: " + e, enableLogging, logFile);
        }
    }

    private static void voidLog(String message, boolean enableLogging, String logFile) {
        if (!enableLogging) return;
        try (FileWriter fw = new FileWriter(logFile, true)) {
            String timestamp = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
            fw.write(timestamp + " " + message + "\n");
        } catch (IOException ignored) {}
    }

    private static String safeSub(String s, int max) {
        return s.length() > max ? s.substring(0, max) : s;
    }
}

File: ApiCall/OllamaAPI.js
const fs = require('fs');
const https = require('https');
const http = require('http');
const { URL } = require('url');

// Default parameters (can be overridden by environment variables or CLI args)
const hostApi = process.env.HOST_API || "http://localhost:11434";
const model = process.env.MODEL || "phi3:mini";
const prompt = process.env.PROMPT || "What is the capital of France?";
const printRaw = process.env.PRINT_RAW === "true";
const enableLogging = process.env.ENABLE_LOGGING === "true";

const logFile = `${__dirname}/llm.log`;

function writeLog(message) {
    const timestamp = new Date().toISOString().replace('T', ' ').replace('Z', '');
    fs.appendFileSync(logFile, `${timestamp} ${message}\n`);
}

function postJson(url, data, callback) {
    const parsedUrl = new URL(url);
    const isHttps = parsedUrl.protocol === 'https:';
    const options = {
        hostname: parsedUrl.hostname,
        port: parsedUrl.port,
        path: parsedUrl.pathname + parsedUrl.search,
        method: 'POST',
        headers: {
            'Content-Type': 'application/json',
            'Content-Length': Buffer.byteLength(data)
        }
    };
    const reqModule = isHttps ? https : http;
    const req = reqModule.request(options, (res) => {
        let body = '';
        res.on('data', (chunk) => { body += chunk; });
        res.on('end', () => callback(null, res.statusCode, body));
    });
    req.on('error', (e) => callback(e));
    req.write(data);
    req.end();
}

const body = JSON.stringify({ model, prompt });

if (enableLogging) writeLog(`Run: hostApi='${hostApi}', model='${model}', prompt='${prompt}', printRaw=${printRaw}`);

postJson(`${hostApi}/api/generate`, body, (err, statusCode, responseString) => {
    if (err) {
        console.error("ERROR:", err);
        if (enableLogging) writeLog(`ERROR: ${err}`);
        return;
    }
    if (enableLogging) writeLog(`STATUS CODE: ${statusCode}`);
    console.log(`STATUS CODE: ${statusCode}`);

    let fullResponse = '';
    const lines = responseString.split('\n');
    for (const line of lines) {
        if (line.trim()) {
            try {
                const json = JSON.parse(line);
                if (json.response) fullResponse += json.response;
            } catch {}
        }
    }
    if (printRaw) {
        console.log(responseString);
        if (enableLogging) writeLog(`RAW RESPONSE: ${responseString.substring(0, 200)}...`);
    } else {
        console.log(fullResponse);
        if (enableLogging) writeLog(`FULL RESPONSE: ${fullResponse.substring(0, 200)}...`);
    }
});